File: ios/DriversTest/DriversTest/view/screen/StatePickerScreen.swift
import SwiftUI

struct StatePickerScreen: View {
    @ObservedObject var vm: QuizViewModel
    @ObservedObject var localizer: Localizer

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                LanguageBarView(localizer: localizer)
                    .padding(.bottom, 4)

                Text(localizer.localized("appTitle"))
                    .font(.system(size: 28, weight: .bold))
                    .foregroundColor(AppTheme.blue)

                Text(localizer.localized("selectStateDesc"))
                    .font(.system(size: 15))
                    .foregroundColor(AppTheme.gray)
                    .padding(.bottom, 8)

                ForEach(vm.allStates) { state in
                    StateCard(state: state, localizer: localizer) {
                        if state.hasQuestions {
                            vm.selectState(state)
                        }
                    }
                }
            }
            .padding(.horizontal, 16)
            .padding(.top, 40)
        }
        .background(AppTheme.grayLight)
    }
}

struct StateCard: View {
    let state: StateInfo
    @ObservedObject var localizer: Localizer
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack {
                VStack(alignment: .leading, spacing: 2) {
                    Text(state.name)
                        .font(.system(size: 17, weight: .semibold))
                        .foregroundColor(.primary)

                    Text(passingText)
                        .font(.system(size: 13))
                        .foregroundColor(AppTheme.gray)
                }

                Spacer()

                if state.hasQuestions {
                    Text(localizer.localized("questionsAvailable", substitutions: ["count": "\(state.totalQuestions)"]))
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundColor(AppTheme.blue)
                } else {
                    Text(localizer.localized("comingSoon"))
                        .font(.system(size: 11))
                        .foregroundColor(AppTheme.gray)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 2)
                        .background(AppTheme.grayLight)
                        .clipShape(RoundedRectangle(cornerRadius: 10))
                }
            }
            .padding(16)
            .cardStyle()
            .opacity(state.hasQuestions ? 1 : 0.5)
        }
        .disabled(!state.hasQuestions)
    }

    private var passingText: String {
        "\(state.agency) \u{00B7} " + localizer.localized("passingScore", substitutions: [
            "pass_pct": "\(state.passingScorePct)",
            "pass_count": "\(state.passCount)",
            "test_count": "\(state.testQuestionCount)",
        ])
    }
}

File: android/app/src/main/java/com/drivers/test/view/screen/StatePickerScreen.kt
package com.drivers.test.view.screen

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.drivers.test.model.StateInfo
import com.drivers.test.theme.AppTheme
import com.drivers.test.view.components.LanguageBar
import com.drivers.test.viewmodel.QuizViewModel

@Composable
fun StatePickerScreen(vm: QuizViewModel) {
    val c = AppTheme.colors
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
            .padding(top = 40.dp),
    ) {
        LanguageBar(vm.currentLang, langLabels = vm.localizer.langLabels) { vm.switchLang(it) }

        Spacer(Modifier.height(16.dp))

        Text(
            vm.t("appTitle"),
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            color = c.blue,
            modifier = Modifier.fillMaxWidth(),
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(4.dp))
        Text(
            vm.t("selectStateDesc"),
            fontSize = 15.sp,
            color = c.gray,
            modifier = Modifier.fillMaxWidth(),
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(24.dp))

        vm.allStates.forEach { state ->
            StateCard(state, vm)
            Spacer(Modifier.height(10.dp))
        }
    }
}

@Composable
private fun StateCard(
    state: StateInfo,
    vm: QuizViewModel,
) {
    val c = AppTheme.colors

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(c.card)
            .border(2.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(12.dp))
            .alpha(if (state.hasQuestions) 1f else 0.5f)
            .clickable(enabled = state.hasQuestions) { vm.selectState(state) }
            .padding(16.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(modifier = Modifier.weight(1f)) {
            Text(state.name, fontSize = 17.sp, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(2.dp))
            Text(
                "${state.agency} \u00B7 " + vm.t(
                    "passingScore",
                    mapOf(
                        "pass_pct" to "${state.passingScorePct}",
                        "pass_count" to "${state.passCount}",
                        "test_count" to "${state.testQuestionCount}",
                    ),
                ),
                fontSize = 13.sp,
                color = c.gray,
            )
        }
        if (state.hasQuestions) {
            Text(
                vm.t("questionsAvailable", mapOf("count" to "${state.totalQuestions}")),
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                color = c.blue,
            )
        } else {
            Text(
                vm.t("comingSoon"),
                fontSize = 11.sp,
                color = c.gray,
                modifier = Modifier
                    .clip(RoundedCornerShape(10.dp))
                    .background(c.grayLight)
                    .padding(horizontal = 8.dp, vertical = 2.dp),
            )
        }
    }
}
